src/kbmod/run_search.py

```diff
@@ -377,49 +377,61 @@ def _calc_barycentric_corr(self, img_info, dist):
 
         # convert this grid to barycentric x,y,z, assuming distance r
         # [obs_to_bary_wdist()]
-        with solar_system_ephemeris.set("de432s"):
-            obs_pos = get_body_barycentric("earth", Time(mjdlist[0], format="mjd"))
+        obs_pos = obs_pos_list[0]
         cobs.representation_type = "cartesian"
         # barycentric distance of observer
-        r2_obs = obs_pos.x * obs_pos.x + obs_pos.y * obs_pos.y + obs_pos.z * obs_pos.z
+        r2_obs = obs_pos.dot(obs_pos)
+        # r2_obs = obs_pos.x * obs_pos.x + obs_pos.y * obs_pos.y + obs_pos.z * obs_pos.z
         # calculate distance r along line of sight that gives correct
         # barycentric distance
         # |obs_pos + r * cobs|^2 = dist^2
         # obs_pos^2 + 2r (obs_pos dot cobs) + cobs^2 = dist^2
-        dot = obs_pos.x * cobs.x + obs_pos.y * cobs.y + obs_pos.z * cobs.z
+        # dot = obs_pos.x * cobs.x + obs_pos.y * cobs.y + obs_pos.z * cobs.z
+        dot = obs_pos.dot(cobs.cartesian)
         bary_dist = dist * u.au
         r = -dot + np.sqrt(bary_dist * bary_dist - r2_obs + dot * dot)
         # barycentric coordinate is observer position + r * line of sight
-        cbary = SkyCoord(
-            obs_pos.x + r * cobs.x,
-            obs_pos.y + r * cobs.y,
-            obs_pos.z + r * cobs.z,
-            representation_type="cartesian",
-        )
+        cbary = astroCoords.SkyCoord(obs_pos + r * cobs.cartesian, representation_type="cartesian")
 
+        return run_search._calculate_barycoeff_list(xlist, ylist, wcslist, cbary, obs_pos_list)
```

Comment:
small nitpick but it's probably more pythonic to have this be
```suggestion
        return self._calculate_barycoeff_list(xlist, ylist, wcslist, cbary, obs_pos_list)
```

Reply:
Changed. I also had to change the related signatures so I'll not commit the suggestion but commit a change. I don't know how else to do that.